eng/common/scripts/artifact-metadata-parsing.ps1

```diff
@@ -0,0 +1,469 @@
+$VERSION_REGEX = "(?<major>\d+)(\.(?<minor>\d+))?(\.(?<patch>\d+))?((?<pre>[^0-9][^\s]+))?"
+$SDIST_PACKAGE_REGEX = "^(?<package>.*)\-(?<versionstring>$VERSION_REGEX$)"
+
+# Posts a github release for each item of the pkgList variable. SilentlyContinue
+function CreateReleases($pkgList, $releaseApiUrl, $releaseSha) {
+  foreach ($pkgInfo in $pkgList) {
+    Write-Host "Creating release $($pkgInfo.Tag)"
+
+    $releaseNotes = ""
+    if ($pkgInfo.ReleaseNotes[$pkgInfo.PackageVersion].ReleaseContent -ne $null) {
+      $releaseNotes = $pkgInfo.ReleaseNotes[$pkgInfo.PackageVersion].ReleaseContent
+    }
+
+    $isPrerelease = $False
+    if ($pkgInfo.PackageVersion -match $VERSION_REGEX) {
+      $preReleaseLabel = $matches["pre"]
+      $isPrerelease = ![string]::IsNullOrEmpty($preReleaseLabel)
+    }
+
+    $url = $releaseApiUrl
+    $body = ConvertTo-Json @{
+      tag_name         = $pkgInfo.Tag
+      target_commitish = $releaseSha
+      name             = $pkgInfo.Tag
+      draft            = $False
+      prerelease       = $isPrerelease
+      body             = $releaseNotes
+    }
+
+    $headers = @{
+      "Content-Type"  = "application/json"
+      "Authorization" = "token $($env:GH_TOKEN)"
+    }
+
+    FireAPIRequest -url $url -body $body -headers $headers -method "Post"
+  }
+}
+
+function FireAPIRequest($url, $method, $body = $null, $headers = $null) {
```

Comment:
To help folks when they read this I'd rename this function `Invoke-RestMethodWithRetries`. Actually Invoke-RestMethod (https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod?view=powershell-7) already supports a retry parameter that we should probably use here and then this method simply becomes a wrapper for handling exceptions. Perhaps `Invoke-RestMethodWithHandling`?

Reply:
Yeah this is fair. Given that I'm not getting this in this release, I'll pull in that refactor as well.